src/parsers/power_models_data.jl

```diff
@@ -162,6 +162,95 @@ function _get_pm_3w_name(
     return "$(get_name(bus_primary))-$(get_name(bus_secondary))-$(get_name(bus_tertiary))-i_$ckt"
 end
 
+"""
+Parses ITC data from a dictionary and constructs a lookup table
+of piecewise linear scaling functions.
+"""
+function _impedance_correction_table_lookup(data::Dict)
+    lookup = Dict{String, PiecewiseLinearData}()
+    type_lookup = Dict{String, String}()
+
+    @info "Reading Impedance Correction Table data"
+    if !haskey(data, "impedance_correction")
+        @info "There is no Impedance Correction Table data in this file"
+        return
+    end
+
+    for (table_number, table_data) in data["impedance_correction"]
+        angle_or_taps_dict = table_data["tap_or_angle"]
+        scaling_factors_dict = table_data["scaling_factor"]
+
+        # Sort dict by key "Fi" or "Ti"
+        indices = sort(
+            parse.(Int, replace.(collect(keys(angle_or_taps_dict)), r"[A-Z]" => "")),
+        )
+
+        x = [
+            angle_or_taps_dict["T$(i)"] for
+            i in indices if haskey(angle_or_taps_dict, "T$(i)")
+        ]
+        y = [
+            scaling_factors_dict["F$(i)"] for
+            i in indices if haskey(scaling_factors_dict, "F$(i)")
+        ]
+
+        if length(x) == length(y)
+            lookup[table_number] =
+                PiecewiseLinearData([(x[i], y[i]) for i in eachindex(x)])
+            table_type =
+                (x[1] >= 0.5 && x[1] <= 1.5) ? "turns_ratio" : "phase_shift_angle"
+            type_lookup[string(table_number)] = table_type
+        else
+            @warn "Impedance correction mismatch at table $table_number: tap and scale count differs"
```

Comment:
I'm not super familiar with the conventions on warning vs. erroring in this parser, but shouldn't this at least be an `@error`, if not a `throw(error)`?